graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/searchtypes/pivot/buckets/OSValuesHandler.java

```diff
@@ -101,6 +105,30 @@ private Script scriptForPivots(Collection<String> pivots) {
         return new Script(scriptSource);
     }
 
+    private TermsAggregationBuilder applyOrdering(Pivot pivot, TermsAggregationBuilder terms, List<BucketOrder> ordering, OSGeneratedQueryContext queryContext) {
+        return sortsOnNumericPivotField(pivot, queryContext)
+                .map(pivotSort -> terms
+                        .subAggregation(AggregationBuilders.max(SORT_HELPER).field(pivotSort.field()))
+                        .order(BucketOrder.aggregation(SORT_HELPER, SortSpec.Direction.Ascending.equals(pivotSort.direction()))))
+                .orElseGet(() -> terms
+                        .order(ordering.isEmpty() ? List.of(DEFAULT_ORDER) : ordering));
+    }
+
+    private Optional<PivotSort> sortsOnNumericPivotField(Pivot pivot, OSGeneratedQueryContext queryContext) {
+        return Optional.ofNullable(pivot.sort())
+                .filter(sorts -> sorts.size() == 1)
```

Comment:
We can have more than one sorts in a pivot, i.e. : 

![image](https://github.com/user-attachments/assets/8e8b5e53-7483-4f02-88db-2ac34c04b7f1)

We are not fixing the sort problem because `sorts.size() == 1` filter.

Reply:
Thanks, good catch! I will fix this and the streams issue before we continue.